util/bb_oscompat.c

```diff
@@ -66,7 +66,10 @@ void comdb2_getservbyname(const char *name, const char *proto, short *port)
 {
     struct servent result_buf, *result = NULL;
     char buf[1024];
-#   if defined(_LINUX_SOURCE)
+#   if defined(_DARWIN_C_SOURCE) // Should be first, as _LINUX_SOURCE is
```

Comment:
Thanks for the pull request! _DARWIN_C_SOURCE is optional on MacOS so it can't be used to reliably detect the OS. Can you please change it to __APPLE__?